gems/sorbet-runtime/lib/types/types/typed_enumerable.rb

```diff
@@ -124,7 +124,7 @@ def describe_obj(obj)
         inferred_val = type_from_instances(obj.values)
         T::Hash[inferred_key, inferred_val]
       when Range
-        T::Range[type_from_instances([obj.first, obj.last])]
+        T::Range[type_from_instances([obj.begin, obj.end])]
```

Comment:
I think this might also be problematic, since `Range.new(1, nil)` would result in being typed as `T::Range[T.nilable(Integer)]` due to the fact that `obj.end` would be `nil`, which is not what you would really expect.